content/utilities/tiny-leaderboard/plugin.js

```diff
@@ -0,0 +1,171 @@
+
+function formatNumber(num, decimalCount = 1) {
+	num = parseInt(num);
+	if (num < 1000) return num;
+	if (num < 1000000) return (num / 1000).toFixed(decimalCount) + "k";
+	if (num < 1000000000) return (num / 1000000).toFixed(decimalCount) + "m";
+	if (num < 1000000000000) return (num / 1000000000).toFixed(decimalCount) + "b";
+	return (num / 1000000000000, 2).toFixed(decimalCount) + "t";
+}
+
+async function downloadLeaderboard() {
+	return fetch('https://api.zkga.me/leaderboard')
+		.then(response => response.json())
+}
+
+// return example: 'hsl(285,100%,70%)'
+function getPlayerColor(ethAddress) {
+	return df.getProcgenUtils().getPlayerColor(ethAddress);
+}
+
+function Plugin() {
+	var o = {};
+	o.div;
+	o.div_timer;
+	o.div_playerList;
+	o.updateTimerInterv = null;
+	o.updateLeaderboardInterv = null;
+	o.firstRender = true;
+	o.leaderboard = [];
+
+	o.init = function () {
+		o.updateTimerInterv = setInterval(o.updateTimer, 1000);
+		o.updateInterv = setInterval(o.updateLeaderboard, 1000 * updateTimeInSeconds);
+		o.updateTimer();
+		o.updateLeaderboard();
+	}
+
+	o.render = function (container) {
+		o.container = container;
+		o.container.style.width = '250px';
+
+		o.div_timer = document.createElement('div');
+		o.div_timer.style.fontSize = "x-large";
+		o.div_timer.style.textAlign = "center";
+		o.div_timer.style["margin-top"] = "-10px";
+		container.appendChild(o.div_timer);
+
+		container.appendChild(document.createElement('hr'));
+
+		o.div_playerList = document.createElement('div');
+		o.div_playerList.style.width = o.container.offsetWidth+"px";
+		container.appendChild(o.div_playerList);
+
+		if (o.firstRender) {
+			o.init();
+			o.firstRender = false;
+		}
+	}
+
+	o.destroy = function () {
+		if (o.updateInterv !== null)
+			clearInterval(o.updateInterv);
+		if (o.updateTimerInterv !== null)
+			clearInterval(o.updateTimer);
+	}
+
+	o.updateTimer = function() {
+		o.div_timer.innerText = getStrTimeUntilRoundEnds();
+	}
+
+	o.updateLeaderboard = async function () {
+		o.leaderboard = await downloadLeaderboard();
+		o.leaderboard = o.leaderboard.entries;
+		o.leaderboard.sort((p1, p2) => {
+			return p2.score - p1.score;
+		});
+
+		o.div_playerList.innerText = "";
+		let table = document.createElement("table");
+		table.style.width = o.container.offsetWidth+"px";
+		let personalRank = o.getLeaderboardRank(df.account);
+		let i = personalRank - leaderboardRange;
+		if (i < 0) i = 0;
+		let max = personalRank + leaderboardRange;
+		for ( ; i < o.leaderboard.length && i <= max; ++i) {
+			o.addPlayerToBoard(table, o.leaderboard[i], i);
+		}
+		o.div_playerList.appendChild(table);
+	}
+
+	o.getLeaderboardRank = function (ethAddress) {
+		for (var i = 0; i < o.leaderboard.length; ++i) {
+			if (ethAddress === o.leaderboard[i].ethAddress)
+				return i;
+		}
+		return -1;
+	}
+
+	o.addPlayerToBoard = function(table, player, rank) {
+		let name = player.twitter !== null ? player.twitter : player.ethAddress.substr(0, 8);
```

Comment:
```suggestion
		let name = player.twitter !== null && player.twitter !== undefined ? player.twitter : player.ethAddress.substr(0, 8);```
		
For me one of these was undefined, not sure why, so didnt work for me. Could investigate why but this seemed to work for me

Reply:
i think instead of
`player.twitter !== null && player.twitter !== undefined`
we could just use
`player.twitter`
it will return false, if its null or undefined, but i am fine with the double check

Reply:
i just learned about a new operator: '??'
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
in this case we could write:
`let name = player.twitter ?? player.ethAddress.substr(0, 8);`
instead of
`let name = player.twitter !== null && player.twitter !== undefined ? player.twitter : player.ethAddress.substr(0, 8);`
but we should not change it, because it might not be compatible with all browsers